### PR TITLE
PRO-1480 PRO-1477 PRO-1481 PRO-1478 PRO-1482 Sets up version-specific use

### PR DESCRIPTION
**Pull request by abea**
- create-project defaults to creating with the a3-boilerplate (probably to be updated if we change the boilerplate name)
- create-project still lets you create projects with a specified repo with `--boilerplate`
- create-project lets you use the A2 boilerplate with the `--a2` flag.
  - This is different from the 1478 proposal (proposed by me). That proposal used a --version flag, but there are only two versions and one is the default. We can also update this before publishing through the renaming ticket.
-  create-widget, create-module, and create-piece work the same way for A2 projects by looking at the installed version of Apostrophe.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,12 +25,20 @@ To create a new project with the tool:
...
 ```
 
-This will create a local copy of our standard [Apostrophe Boilerplate](https://github.com/apostrophecms/apostrophe-boilerplate).
+This will create a local copy of the [Apostrophe 3 boilerplate](https://github.com/apostrophecms/a3-boilerplate).
 
 ### options
 
+#### `--a2`
+
+Use the `--a2` flag when creating an Apostrophe 2 project to use [the boilerplate](http://github.com/apostrophecms/apostrophe-boilerplate) for that version.
```
Comment (boutell) [inline README.md:34]: Say what the default is explicitly

File: lib/boilerplate.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+const util = require('./util');
+
+module.exports = function (moduleName, options = {}) {
+  return {
+    a2: {
+      'create-module': {
+        moduleConfig: stripIndent`
+          module.exports = {
+            extend: 'apostrophe-module',
+            label: '${util.titleCase(moduleName.replace(/-/g, ' '))}',
```
Comment (boutell) [inline lib/boilerplate.js:11]: Generic modules have no concept of a label option out of the box.

File: lib/boilerplate.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      },
+      'create-piece': {
+        pieceConfig: stripIndent`
+          module.exports = {
+            extend: 'apostrophe-pieces',
+            // TODO: Update the piece type name to be the singular version of the module
+            // name. For example, if the module is "people" the piece type name should be
+            // "person".
+            name: '${moduleName}',
+            // TODO: Update the singular label for editors
```
Comment (boutell) [inline lib/boilerplate.js:26]: I'm assuming there's no regression here.

File: lib/boilerplate.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      'create-widget': {
+        widgetsConfig: stripIndent`
+          module.exports = {
+            extend: 'apostrophe-widgets',
+            label: '${util.titleCase(moduleName.replace(/-/g, ' '))} Widget',
+            addFields: []${options.player ? `,
+            construct: function (self, options) {
+              self.pushAsset('script', 'lean', { when: 'lean' });
+            }`
+              : ''}
```
Comment (boutell) [inline lib/boilerplate.js:57]: Funky layout.

File: lib/commands/create-piece.js [hunk 5]
```diff
@@ -2,87 +2,83 @@ require('shelljs/global');
...
+        util.log('create-piece', `Creating a ${moduleName}-pages folder with index.js and appropriate views`);
+
+        const pagesConfig = strings.pagesConfig;
+        const pageDir = `${modulePath}-pages`;
+
+        mkdir('-p', path.join(pageDir));
+        fs.writeFileSync(path.join(pageDir, 'index.js'), pagesConfig);
+
+        mkdir('-p', path.join(pageDir, 'views'));
+        exec(`touch ${path.join(pageDir, 'views', 'show.html')}`);
```
Comment (boutell) [inline lib/commands/create-piece.js:59]: If you're going to use `path.join` for correctness, you should also be using shell-quote in all of your exec calls that involve user input:

https://github.com/substack/node-shell-quote

Comment (boutell) [reply lib/commands/create-piece.js:59]: Or, you could take an approach based on `spawnSync` which would inherently avoid the problem of misquoted or unquoted command line parameter strings with spaces, punctuation, etc.

Comment (boutell) [reply lib/commands/create-piece.js:59]: Since you seem to only be doing that for `touch`, you could just use the `touch` npm module.

**Review by boutell — CHANGES_REQUESTED** [review]

File: lib/boilerplate.js (hunk 3 above)
Comment (abea) [reply lib/boilerplate.js:26]: What would the regression be?

**Review by boutell — APPROVED** [review]

**Pull request merged**